hs628_tools/PCA_refinement.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jul 21 14:43:54 2018
+
+@author: BlauBear
+"""
+
+import numpy as np
+import pandas as pd
+from sklearn.cross_validation import cross_val_score, ShuffleSplit
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+
+#find out which variables are correlated
+#X is the independent variables, Y is the dependent variables, names is the variable names
+def correlated_variables (X, Y, names):
+    """"This function finds out which variables are correlated
+        Args: 
+            X: independent variables
+            Y: dependent variables
+            names: variable names
+        Returned: 
+            scores: list of variable names and their correlation scores
+    """"
```

Comment:
Given the limited amount of time left, the clique analysis might be a bit too involved.  It might work out reasonably to just check the pairwise correlations from numpy.corrcoef, make a set from all the features that are correlated to any other above your threshold, and then rely on the PCA to effectively handle the grouping aspect.